CMakeLists.txt

```diff
@@ -565,3 +575,121 @@ add_subdirectory (utils)
 include (cmake/print_include_directories.cmake)
 
 include (cmake/sanitize_target_link_libraries.cmake)
+
+# Various build speed optimizations.
+option(FASTBUILD_CONTRIB_UNITY "Use unity builds on the contrib/ folder. (No-op on CMake < 3.16.0)" ON)
+option(FASTBUILD_FUNCTIONS_UNITY "Use unity builds on the src/Functions folder. (No-op on CMake < 3.19.0)" ON)
+option(FASTBUILD_NO_PUBNAMES "Do not emit DWARF pubnames when using Clang. Slows GDB symbol loading. Reduces IO." OFF)
+option(FASTBUILD_SPLIT_DWARF "Build with split DWARF files. Makes debugging symbols non-portable. Reduces IO." OFF)
+
+# Check Unity build compatibility.
+if ((FASTBUILD_CONTRIB_UNITY OR FASTBUILD_FUNCTIONS_UNITY) AND CMAKE_VERSION VERSION_LESS "3.16.0")
+    message("Disabling Unity builds. (Needs CMake >= 3.16.0)")
+    set(FASTBUILD_CONTRIB_UNITY OFF CACHE BOOL INTERNAL FORCE)
```

Comment:
I would consider changing this message's status to "warning" (and also statuses of all CMake-related optimizations messages) because, unlike ccache and other external tools, CMake is mandatory to this project and users should be aware that the old versions would provide worse build times / binary size.